<!-- -->
<launch>
    <param name="use_sim_time" value="true" />   <!--if true, the package will generate clock itself -->
    <node pkg="tf" type="static_transform_publisher" name="world_to_odominit" args="0 0 1 0 0 -1.57 /world /odom_init 10" /> 
    <node pkg="pop_planar_slam" type="planar_test" name="planar_test" output="screen">	
        <rosparam file="$(find pop_planar_slam)/params/plane_3d_tum_far.yaml" command="load"/>
	<remap from="/image" to ="/mono/image_mono"/>  <!--/image_raw -->
	<remap from="/camera_info" to ="/mono/camera_info"/>  <!--/camera_info -->
    </node>
    
    <rosparam file="$(find pop_up_wall)/params/popup_tum_far.yaml" command="load"/>

    # used in tum rgbd dataset, which provides the ground truth pose in tf format, I need to extract it to odometry pose.
    <node pkg="odom_tf_utils" type="tf_to_odom" name="tf_to_odom" output="screen">
	<param name="odom_topic" value="/truth_pose" />
	<param name="source_frame" value ="/world"/>  <!--/image_raw -->
	<param name="target_frame" value ="/openni_rgb_optical_frame"/>  <!--/camera_info -->
    </node>


    <!-- play bag file -->
    <node pkg="rosbag" type="play" name="rosbag" args="-r 2 -s 0 -d 3
	  $(find pop_planar_slam)/dataset/tum_far_preproc_tf.bag --clock">
    </node>
	
    <!-- launch rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pop_planar_slam)/rviz/pop_slam.rviz"/>	
	
</launch>
